evalml/automl/engine/cf_engine.py

```diff
@@ -0,0 +1,161 @@
+from evalml.automl.engine.engine_base import (
+    EngineBase,
+    EngineComputation,
+    evaluate_pipeline,
+    score_pipeline,
+    train_pipeline,
+)
+
+
+class CFClient:
```

Comment:
Can we get rid of this class? The `ThreadPoolExecutor` and `ProcessPoolExecutors` already have context management and submit method in their api.

https://docs.python.org/3/library/concurrent.futures.html#threadpoolexecutor-example